### ISSUE TITLE
Update changelog view UI to use objectdata_v2

### ISSUE THREAD

**Issue by glennmatthews** [nautobot/nautobot#1530]
### As ...

Nelly - Network Engineer

### I want ...

To get a detailed and clear view of any changes that were made to a Nautobot object.

### So that ...

I can easily see exactly what has changed without needing to refer to different/additional views to translate UUID PKs into object names, etc.

### I know this is done when...

The changelog detail view for an object is based on the REST API serialization (`objectdata_v2`, as added in #1389) rather than the shallow dictionary serialization (`objectdata`), if available for a given changelog.

If `objectdata_v2` isn't available, the view should gracefully fail back to the current `objectdata` behavior.

### Optional - Feature groups this request pertains to.

- [ ] Automation
- [ ] Circuits
- [ ] DCIM
- [ ] IPAM
- [ ] Misc (including Data Sources)
- [ ] Organization
- [ ] Plugins (and other Extensibility)
- [ ] Security (Secrets, etc)
- [ ] Image Management
- [X] UI/UX
- [ ] Documentation
- [ ] Other (not directly a platform feature)

### Database Changes

None, just using the existing DB changes made in #1389.

### External Dependencies

None.

**Comment by bryanculver** [comment]
@nautobot/core was this resolved with 1.3, 1.4, and https://github.com/nautobot/nautobot/issues/2795?

**Comment by glennmatthews** [comment]
I don't *think* so - IIRC the "differences" view was updated in #2795/#2861 to use either format, but the `Object Data` view still always uses the v1 data. For example:

![image](https://user-images.githubusercontent.com/5603551/217840534-d777c97c-dc73-4118-bcda-5ee0468e112f.png)

note that the diff shows a change in the field `compliance_config`, which is present in the v2 (REST API) serialization but not in the v1 serialization - and therefore this field is missing from the "Object Data" area entirely.

**Comment by gsnider2195** [comment]
> I don't _think_ so - IIRC the "differences" view was updated in #2795/#2861 to use either format, but the `Object Data` view still always uses the v1 data. For example:

This is correct. The `Difference` panel uses `object_data_v2` if it's available in both the before/after data and only falls back to `object_data_v1` for legacy data that was generated before v2 was implemented. Agree that the same type of logic should be implemented for the `Object Data` panel.
